apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx

```diff
@@ -1,49 +1,166 @@
+  const handleAiEnabledChange = idToUpdate => {
+    const newLearningGoalData = learningGoalList.map(learningGoal => {
+      if (idToUpdate === learningGoal.id) {
+        const updatedAiValue = !learningGoal.aiEnabled;
+        return {
+          ...learningGoal,
+          ['aiEnabled']: updatedAiValue,
+        };
+      } else {
+        return learningGoal;
+      }
+    });
+    setLearningGoalList(newLearningGoalData);
+  };
 
-    setCurrentId(updatedId);
+  const handleLearningGoalNameChange = (updatedName, idToUpdate) => {
+    const newLearningGoalData = learningGoalList.map(learningGoal => {
+      if (idToUpdate === learningGoal.id) {
+        return {
+          ...learningGoal,
+          ['learningGoal']: updatedName,
+        };
+      } else {
+        return learningGoal;
+      }
+    });
+    setLearningGoalList(newLearningGoalData);
+  };
 
-    // update the goal list
-    setLearningGoalList(oldList => {
-      return [...oldList, {id: updatedId}];
+  // TODO: Check that there is at least one programming level here
+  const initialLevelForAssessment = !!rubric ? rubric.levelId : levels[0].id;
+  const [selectedLevelForAssessment, setSelectedLevelForAssessment] = useState(
+    initialLevelForAssessment
+  );
+
+  const saveRubric = event => {
+    event.preventDefault();
+    const dataUrl = !!rubric ? `/rubrics/${rubric.id}/edit` : RUBRIC_PATH;
+    const method = !!rubric ? 'PATCH' : 'POST';
+
+    // Checking that the csrf-token exists since it is disabled on test
+    const csrfToken = document.querySelector('meta[name="csrf-token"]')
+      ? document.querySelector('meta[name="csrf-token"]').attributes['content']
+          .value
+      : null;
+
+    const learningGoalListAsData = transformKeys(learningGoalList);
+
+    const rubric_data = {
+      levelId: selectedLevelForAssessment,
+      lessonId: lessonId,
+      learningGoalsAttributes: learningGoalListAsData,
+    };
+
+    fetch(dataUrl, {
+      method: method,
+      headers: {
+        'Content-Type': 'application/json',
+        'X-CSRF-Token': csrfToken,
+      },
+      body: JSON.stringify(rubric_data),
+    })
+      .then(response => {
+        let redirectUrl = response.url;
+        navigateToHref(redirectUrl);
+      })
+      .catch(err => {
+        console.error('Error saving rubric:' + err);
+      });
+  };
+
+  function transformKeys(startingList) {
+    let newList = [];
+
+    startingList.forEach(item => {
+      let newItem = {};
+      for (const [key, value] of Object.entries(item)) {
+        newItem[snakeCase(key)] = value;
+      }
+      newList.push(newItem);
     });
+
+    return newList;
+  }
+
+  const handleDropdownChange = event => {
+    setSelectedLevelForAssessment(event.target.value);
   };
 
   // TODO: In the future we might want to filter the levels in the dropdown for "submittable" levels
   //  "submittable" is in the properties of each level in the list.
   return (
-    <div>
+    <form id={FORM_ID}>
```

Comment:
Why is this needed? (this may also be the reason for all of the weird button behavior)